File: src/main/java/top/zbeboy/isy/glue/util/ResultUtils.java
package top.zbeboy.isy.glue.util;

import lombok.Data;
import lombok.extern.slf4j.Slf4j;

/**
 * Created by lenovo on 2017-03-28.
 */
@Slf4j
@Data(staticConstructor = "of")
public class ResultUtils<T> {

    private T data;
    private long totalElements;

    public ResultUtils<T> data(T data) {
        this.data = data;
        return this;
    }

    public ResultUtils<T> totalElements(long totalElements) {
        this.totalElements = totalElements;
        return this;
    }
}

File: src/main/java/top/zbeboy/isy/glue/util/ResultUtils.kt
package top.zbeboy.isy.glue.util

/**
 * Created by zbeboy 2017-10-31 .
 **/
class ResultUtils<T> {
    private var data: T? = null
    private var totalElements: Long = 0

    constructor()

    constructor(data: T?, totalElements: Long) {
        this.data = data
        this.totalElements = totalElements
    }

    fun data(data: T): ResultUtils<T> {
        this.data = data
        return this
    }

    fun totalElements(totalElements: Long): ResultUtils<T> {
        this.totalElements = totalElements
        return this
    }

    fun getData(): T? {
        return this.data
    }

    fun getTotalElements(): Long {
        return this.totalElements
    }
}